extensions/devservices/mssql/src/main/java/io/quarkus/devservices/mssql/deployment/MSSQLDevServicesProcessor.java

```diff
@@ -25,7 +25,24 @@ public RunningDevServicesDatasource startDatabase(Optional<String> username, Opt
                         imageName.orElse(MSSQLServerContainer.IMAGE + ":" + MSSQLServerContainer.DEFAULT_TAG))
                                 .withPassword(password.orElse("Quarkuspassword1"));
                 container.start();
-                return new RunningDevServicesDatasource(container.getJdbcUrl(), container.getUsername(),
+                StringBuilder additionalArgs = new StringBuilder();
+                for (Map.Entry<String, String> i : additionalProperties.entrySet()) {
+                    if (additionalArgs.length() > 0) {
+                        additionalArgs.append("&");
+                    }
+                    additionalArgs.append(i.getKey());
+                    additionalArgs.append("=");
+                    additionalArgs.append(i.getValue());
+                }
+                String jdbcUrl = container.getJdbcUrl();
+                if (additionalArgs.length() > 0) {
+                    if (jdbcUrl.contains("?")) {
```

Comment:
MSSQL is actually a bit special in this regard. The parameters are not separated using `?` and `&`. Please have a look at https://github.com/testcontainers/testcontainers-java/blob/master/modules/mssqlserver/src/main/java/org/testcontainers/containers/MSSQLServerContainer.java#L94-L95.

Reply:
@knutwannheden would you be willing to prepare a PR to adjust that?

Reply:
Yes, sure.